keps/349-all-or-nothing/README.md

```diff
@@ -345,12 +363,64 @@ condition, so the corresponding job is unsuspended without further waiting.
 
 ### Timeout on reaching the PodsReady condition
 
-We introduce a timeout, defined in the `waitForPodsReady.timeoutSeconds` field, on reaching the `PodsReady` condition since the job
-is unsuspended (the time of unsuspending a job is marked by the Job's
-`job.status.startTime` field). When the timeout is exceeded, the Kueue's Job
+
+We introduce two timeouts defined in the `waitForPodsReady.timeout` and `waitForPodsReady.recoveryTimeout` fields.
+
+First one applies before the job has started. It tracks the time between job getting unsuspended for the first time (the time of unsuspending a job is marked by the Job's
+`job.status.startTime` field) and reaching the `PodsReady=true` condition (marked by condition's `.lastTransitionTime`).
+
+```mermaid
+flowchart TD;
+	start@{ shape: f-circ};
+	id1(Suspended=true);
+	id2("PodsReady=false
+	waits for .timeoutSeconds");
+	id3(PodsReady=true);
+	id4("Suspended=true (Requeued)");
+
+	start--Workload gets admitted-->id1;
+	id1 --> id2;
+
+	id2 --"Doesn't exceed the timeout" --> id3
+	id2 --"Exceeds the timeout" --> id4
+```
+
+
+Second one applies when the job has already started and some pod failed while the job is running. It tracks the time between changing `PodsReady` condition to `false` and reaching the
+`PodsReady=true` condition once again.
+
+```mermaid
+flowchart TD;
+	start@{ shape: f-circ};
+	id1(Suspended=true);
+	id2("PodsReady=false(1st)");
+	id3(PodsReady=true);
+	id4("PodsReady=false(2nd)
+	waits for
+	.recoveryTimeout");
+	id5("Suspended=true (Requeued)");
+
+
+	start--Workload gets admitted-->id1;
+	id1 --> id2;
+
+	id2 --"Job started" --> id3
+	id3 --"Pod failed"--> id4
+	id4 --"Pod recovered"--> id3
+	id4 --"timeout	exceeded"--> id5
+```
+
+We introduce new `WorkloadWaitForPodsStart` and `WorkloadWaitForPodsRecovery` reasons to distinguish the reasons of setting the `PodsReady=false` condition.
```

Comment:
How can existing users migrate the previous "PodsReady" reason to the new reason?
Is there any migration plans?